Develop

Aldatrualda opened the pull request:
I am waiting for your comments to start working on "Show" methods.

Commit:
+ Added the class SQLConnection
class SQLConnection:
+ Added the static variable ConnectionToSQL
+ Added the static string method SelectPersonalPlayerIdFromPersonalDetails

+ Added Employee Page Form
+ Added Employee Page class

+ Added a new label to each page form "Required fields"

Commit:
+ Added a method checking if some boxes are empty "checkIfRequiredBoxesAreEmpty" to each page
+ Added methods closing a page if click on the "Back" button

Commit:
+- Changed the name of a class from SQLConnection to SQLVariableContainer
+ Added a class "Generate Password" using in "EmployeePage" so far
+ Added method to show Employee's login and password in a message box

# modified birthhday variable in "Employee" and "Player" classes

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+                Surname = textBoxSurname.Text,
+                AdditionalName = textBoxAdditionalName.Text,
+                PhoneNumber = textBoxPhoneNumber.Text
+            };
```
Можно сразу в 39 строке прописать, без создания лишний переменной + по логике ты сначала создаешь объект (да без записи в БД), а потом делаешь проверку на поля.

Aldatrualda replied:
Согласен, исправил.

Я сначала теперь проверяю, а потом только создаю объект. Без лишней переменной.

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/1ecaadfb-c933-4597-bc6d-8ba51eebdd3e)

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+        {
+            GeneratePassword generatePassword = new GeneratePassword();
+            return generatePassword.Password;
+        }
```
Можно уменьшить до одной строчки. Опять создаешь лишнюю переменную. По хорошему этот метод сделать статическим.

Aldatrualda replied:
Исправляю, нравится мне создавать лишние объекты...

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/7415e818-1457-4650-a1bd-af9ccb8e025c)

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Something went wrong", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
```
Я бы оставил одну кнопку, а то с двумя ты как будто имеешь разную логику в зависимости от нажатия.

Aldatrualda replied:
Так тут одна кнопка OK, другое это икона.

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+            };
+
+            if (!CheckRequiredBoxesAreEmpty())
+            {
```
Сюда добавь логирование: создание пользователя. И отображение парочки атрибутов.

Aldatrualda replied:
Принял

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+        private void InsertDataIntoPersonalDetails(Employee employee, NpgsqlConnection connection)
+        {
+            string insertQuery = "INSERT INTO personal_details(name, surname, additional_name, birthday, phone_number) " +
+                "VALUES ('" + employee.Name + "', '" + employee.Surname + "', '" + employee.AdditionalName + "', '" + employee.Birthday + "', '" + employee.PhoneNumber + "')";
```
Используй интерполяцию строк

Aldatrualda replied:
Исправил.

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/885e4ec0-7d3c-46a4-9569-a4cf216a8be3)

Aldatrualda replied:
Убрал первый знак доллара :)

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+        private void InsertDataIntoEmployeeOfRealMadrid(Employee employee, NpgsqlConnection connection)
+        {
+            string insertQuery = "INSERT INTO employee_of_real_madrid(role_access, personal_employee_details, login, password) " +
+                "VALUES ('" + employee.RoleAccessModifier + "', " + SQLVariableContainer.SelectPersonalPlayerIdFromPersonalDetails + ", '"
```
Здесь и во всех последующих местах тоже

DenisPushka commented on RealMadridDesktopApplication/Forms/AddEmployeePage.cs:
```diff
@@ -0,0 +1,123 @@
+        /// <summary>
+        /// Name, Surname, Role, and Phone Number are requierd fields
+        /// </summary>
+        private bool CheckRequiredBoxesAreEmpty()
```
Опиши returns по типу: true тогда то тогда, иначе то то

DenisPushka commented on RealMadridDesktopApplication/Forms/AddParentPage.cs:
```diff
@@ -21,30 +22,57 @@ public AddParentPage()
             string insertQuery = "INSERT INTO parent_of_player(name, surname, phone_number) " +
-                                 "VALUES ( '" + parent.Name + "', '" + parent.Surname + "', '" + parent.PhoneNumber + "')";
-            using (NpgsqlCommand command = new NpgsqlCommand())
+                                 "VALUES ('" + parent.Name + "', '" + parent.Surname + "', '" + parent.PhoneNumber + "')";
```
интерполяция

Aldatrualda replied:
Исправил :)

DenisPushka commented on RealMadridDesktopApplication/Modules/Employee.cs:
```diff
@@ -26,6 +26,7 @@ public Employee(AccessModifier roleAccessModifier, string login, string password
 
             this.login = login;
             this.password = password;
+            Birthday = birthdayBuffer[2] + "-" + birthdayBuffer[1] + "-" + birthdayBuffer[0];
```
Вынеси цифры в константы и опиши их

Aldatrualda replied:
Исправил.

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/c400ff91-d3c8-4d4a-987f-d381d8b267c4)

DenisPushka commented on RealMadridDesktopApplication/Password/GeneratePassword.cs:
```diff
@@ -0,0 +1,37 @@
+
+    public class GeneratePassword
+    {
+        private string listOfChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!@#$%^&*()";
```
вроде можно сделать константой

Aldatrualda replied:
Можно, исправления не допускаются :)

DenisPushka commented on RealMadridDesktopApplication/Password/GeneratePassword.cs:
```diff
@@ -0,0 +1,37 @@
+            Random random = new Random();
+            for (int i = 0; i < 9; i++)
+            {
+                int randomIndex = random.Next(0, listOfChars.Length);
+                password.Append(listOfChars[randomIndex]);
```
RandomIndex можно убрать, сразу вставляй в индекс

Aldatrualda replied:
Исправил. 

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/81e4ec3a-8c98-4532-bc61-be7f93b7a9c7)

DenisPushka commented on RealMadridDesktopApplication/SQLConnection/SQLVariableContainer.cs:
```diff
@@ -0,0 +1,21 @@
+        /// <summary>
+        /// The info about the local database. I use it to connect to my SQL server.
+        /// </summary>
+        public static string ConnectionToSQL= "Host=localhost;Port=5432;Database=RealMadridDB;Username=postgres;Password=123456";
```
Вынеси в зависимости проекта

Aldatrualda replied:
Ты имеешь ввиду вынести весь класс или только данную переменную?

Commit:
+ Added SelectDataFromEmployeeOfRealMadrid method to SQLConnection package

- Removed SelectDataFromEmployeeOfRealMadrid method from LoginPage

# Fixed mistakes such as - creating unneccessery objects, added comments, added const variables to Employee model,  started using string interpolation.
